src/pngwriter.cc

```diff
@@ -787,23 +795,24 @@ int pngwriter::read(int x, int y, int colour) const
 
 	if(bit_depth_ == 8)
 	  {
+	     int const scale8To16Bit = 257;   // x/255*65535 = x*257
```

Comment:
sounds legit. be aware you reach [0,1] and not float.

self assign: just means I review it :)